main_hall.py

```diff
@@ -0,0 +1,92 @@
+class Main_Hall:
+    floor1_is_flooded = True
+    if floor1_1_flooded = True:
+        def __int__(self,global_state,inventory):
+            self.global_state = global_state
+            self.inventory = inventory
+            self.set_state()
+            self.description = """ You are the end end of senior hall near the library.
+                               To the north there are stairs leading down to the main hall which looks flooded,
+                               to the south there is a door that leads to the ASL room,
+                               to the west is the history hall,
+                               to the east is the senior hall. Choose wisely.. """
+            print(self.description())
+            self.run()
+            
+
+        def run():
+            self.route(input(">>"))
+    
+        def route(command):
+            if command in ["go north", "go south", "go east", "go west"]:
+                move(command.split(" ")[1])
+            else:
+                if command in ["Walk down stairs", "Enter stairs", "pass through stairs,"]:
+                    print("You cannot enter! The water is too deep to pass!")
+                else:
+                    print("What???")
+                self.run()
+
+        def move(direction):
+            for key in self.state.keys():
+                self.global_state[self.name+"_"+key] = self.state[key]
+            if direction == "north":
+                print("The stairs down to the art room seem to be flooded")
+            if direction == "west":
+                room = SeniorHall(self.global_state,self.inventory)
+                room.run()
+            if direction = "south":
+                print("The doors to the ASL room seem to be locked. You look into the window but fail to see anything for it is too dark")
```

Comment:
@aebooth How do we want to handle string wrapping? You have it there wrapping at ~55 or 60 with a 5 space hanging indent. That seems odd. The repeated quotes method I used above leaves it as a single string to be wrapped automatically by the printing function (which I'm fine with), but we may want to manually wrap to 80 characters, in which case something along the lines of this should work well:
```python
code:
    code(42)
    print("""\
123
ABC
""")
    more_code()
```